### ISSUE TITLE
Passing `import.meta.path` to a macro segfaults

### ISSUE THREAD

**Issue by vjpr** [oven-sh/bun#3726]
### What version of Bun is running?

0.7.0

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

```
import {foo} from './macro-test.ts' with {type: 'macro'}
console.log('foo:', foo(import.meta.path))
```

```
export function foo(file) {
  // TODO: Extract AST and return it.
  return file
}
```

### What do you see instead?

Segfaults

### Additional information

I want to use a macro to serialize an AST of the current file to expose to it at runtime.

Seems like `import.meta.path` should be statically analyzable...

Regardless, shouldn't segfault.

**Comment by vjpr** [comment]
Actually, even passing a string to this macro causes a segfault.

---

EDIT: So when its in the module scope, it segfaults. But when I move it inside a function it works.

But `import.meta.path` still segfaults...

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.0.1
